### PR TITLE
Implemented %ASA-3-114006

### LINKED ISSUE #23: Implement %ASA-3-114006: Failed to get port statistics in 4GE SSM I/O card (error error_string).
Requirement is to split up this message type into the following fields (columns):

Type == ASA
Severity == 3
ID == 114006
Text = Failed to get port statistics in 4GE SSM I/O card
Error = error_string

### PR DESCRIPTION
**Pull request by lancenonce**
Submission for team NULL

### CODE REVIEW

**Review by claesmk — CHANGES_REQUESTED** [review]
The way your requirements are written in your [issue](https://github.com/NCATComp410/comp410_spring_2021/issues/23) you are expecting 
```
Error == error_string 
```
not
```
Error == error error_string
```
The reason this is happening is you have nested parenthesis in your regex:
```
m = re.search(r'(error (\w+))', df.loc[id, 'Text'])
```
You can fix this by escaping the open parenthesis like this:
```
m = re.search(r'\(error (\w+)', df.loc[id, 'Text'])
```
You don't really need the closing parenthesis because \w+ won't match that, but, you could include it also with an escape
```
m = re.search(r'\(error (\w+)\)', df.loc[id, 'Text'])
```
regex can be very tricky.  The regex links I included in the code, including regex101.com are very helpful for building and debugging regex. 

Please make this change and push your fork again, your pull request should update automatically.

**Comment by lancenonce** [comment]
Mr. Claes, my tests fail when I make those changes. Are there any more
suggestions you have?

Lance N. Davis

**Comment by claesmk** [comment]
Your tests will fail unless you also change your test case to match the requirement
```
self.assertEqual('error error_string', df.loc[114006, 'Error']) 
```
Needs to be 
```
self.assertEqual('error_string', df.loc[114006, 'Error']
```

**Comment by lancenonce** [comment]
They worked! I pushed the changes. Thank you.

Lance N. Davis

File: id_pkg/test/test_log_parse.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def test_syslog_file(self):
 
         # Open the syslog file
         # https://docs.python.org/3/tutorial/inputoutput.html
-        with open(os.path.join(self.data_path, fname), encoding='utf-8') as f:
```
Comment (claesmk) [inline id_pkg/test/test_log_parse.py:31]: I don't think you updated your branch to latest.  If you commit this you'll break unit test for anyone using a PC

**Review by claesmk — CHANGES_REQUESTED** [review]

**Comment by lancenonce** [comment]
Mr. Claes, I have no idea how to resolve this.

Lance N. Davis

File: id_pkg/test/test_log_parse.py (hunk 1 above)
Comment (lancenonce) [reply id_pkg/test/test_log_parse.py:31]: My branch says it is up to date, should I try again now?

**Comment by claesmk** [comment]
It looks like something didn't merge correctly.  Perhaps you merged main and null to the same branch at different times. I'd suggest making a new branch off of NCAT410/null and add your code there, then push that to your fork, then create a new pull request off that branch:

```
(py39) % git fetch upstream
(py39) % git checkout upstream/null
(py39) % git branch sprint2_2
(py39) % git checkout sprint2_2
Switched to branch ‘sprint2_2’
(py39) % git status
On branch sprint2_2
nothing to commit, working tree clean
(py39) mike.claes@MBP-MikeC id_pkg % git merge upstream/null
Already up to date.
(py39) % cd id_pkg/test
(py39) % grep “with open” *.py
test_log_parse.py:        with open(os.path.join(self.data_path, fname), encoding=‘utf-8’) as f:
```

You can see in this new branch I created the encoding=‘utf-8’ is correct..

**Pull request closed without merging**